sim/pokemon.js

```diff
@@ -873,7 +873,7 @@ class Pokemon {
 		if ((pokemon.transformed && this.battle.gen >= 2) || (this.transformed && this.battle.gen >= 5)) {
 			return false;
 		}
-		if (!this.formeChange(template, null)) {
+		if (!this.formeChange(template)) {
```

Comment:
```suggestion
		if (!this.formeChange(template, null)) {
```

Reply:
... but then you're not setting a source effect for `transformInfo`, which defeats the object. So maybe the right way to go is to separate out some functions of `formeChange` into `setTemplate` which is called by `transformInto`, `clearVolatile` and `formeChange`, leaving the rest in `formeChange`.

Reply:
Oh, I think I misread the code. I didn't realize this was `transformInto` and not `clearVolatile`. My bad. I did say to always set a source effect for `transformInto`, and that was my intent.

Reply:
In hindsight, I do like your new code better, though.